R/ckan_fetch.R

```diff
@@ -89,10 +89,10 @@ ckan_fetch <- function(x, store = "session", path = "file", format = NULL,
 
   store <- match.arg(store, c("session", "disk"))
   file_fmt <- file_fmt(x)
-  if (identical(file_fmt, character(0)) & is.null(format)) {
+  if (is.na(file_fmt) & is.null(format)) {
```

Comment:
hmm, 3 questions:

1. i'd imagine not the best idea to re-assign `file_fmt` function to a variable of the same name (line above this if statement). same question applies to the same usage in `zzz.R` file below
2. do we want `&&` instead of `&` here?
3. `is.na` can give a vector of length > 1, so maybe `all(is.na(file_fmt))`, unless there's absolutely no way `file_fmt` can be > length 1

Reply:
1. good call on `file_fmt` - i wrote that forever ago, but bad practice! have replaced with `derived_file_fmt` throughout.

2. i'm never clear on `&&` versus `&`, but happy to change it!

3. i think that there's no meaningful way for (now named) `derived_file_fmt` to have length > 1 - in theory you *can* pass x with length > 1 to ckan_fetch, but it errors (unrelated to the format, especially since it's explicitly specified):

``` r
library(ckanr)
#> Loading required package: DBI

url <- "https://ckan0.cf.opendata.inter.prod-toronto.ca/datastore/dump/c57c3e1c-20e2-470f-bc82-e39a0264be31"

ckan_fetch(c(url, url), format = "csv")
#> Error in private$crul_h_pool[[z]] <- handle: wrong args for environment subassignment
```

would it make sense to have the function error and stop if x is > length 1? and if so, would you rather this in `ckan_fetch()` or `fetch_GET()`?

Reply:
1. thanks. 
2. `&` is vectorized, whereas `&&` is not. 

```r
foo = letters[1:3]
foo == "a"
#> [1]  TRUE FALSE FALSE
foo == "b"
#> [1] FALSE  TRUE FALSE
foo == "a" & foo == "b"
#> [1] FALSE FALSE FALSE
foo == "a" && foo == "b"
#> [1] FALSE
```

Here we want `&&` because if/else expect a single boolean

3. I assume you designed the function so that `x` handles only 1 url, correct? Maybe use an assertion at the top of the function to fail early if `length(x) > 1`, e.g., `stopifnot("x must be length 1" = length(x) == 1)`

Reply:
thanks, all fixed now! same with the comments below - will mark them resolved now